### PR TITLE
Adjusted Given-When steps in ChecksumContext

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Combined `given-when` steps splitted into seperate functions and check-success assertion added on given steps.

## Related Issue
- Part of https://github.com/owncloud/QA/issues/635

## How Has This Been Tested?

- CI
- :robot:

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/ChecksumContext.php [hunk 1]
```diff
@@ -67,9 +67,40 @@ public function userUploadsFileToWithChecksumUsingTheAPI(
...
+		Assert::assertGreaterThanOrEqual(
+			200,
+			$response->getStatusCode(),
+			'Error response code! Found response: ' . $response
+		);
+		Assert::assertLessThanOrEqual(
+			300,
+			$response->getStatusCode(),
+			'Error response code! Found response: ' . $response
+		);
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/ChecksumContext.php]: You can check all this using `theHTTPStatusCodeShouldBeBetween()`

File: tests/acceptance/features/bootstrap/ChecksumContext.php [hunk 2]
```diff
@@ -67,9 +67,40 @@ public function userUploadsFileToWithChecksumUsingTheAPI(
...
+	public function userHasUploadedFileToWithChecksumUsingTheAPI(
+		$user, $source, $destination, $checksum
+	) {
+		$this->userUploadsFileToWithChecksumUsingTheAPI(
+			$user,
+			$source,
+			$destination,
+			$checksum
+		);
+		$response = $this->featureContext->getResponse();
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/ChecksumContext.php]: remove this line. The suggested function for checking statuscodes handles that.

File: tests/acceptance/features/bootstrap/ChecksumContext.php (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/ChecksumContext.php]: done! :+1:

File: tests/acceptance/features/bootstrap/ChecksumContext.php [hunk 3]
```diff
@@ -22,6 +22,7 @@
 use Behat\Behat\Context\Context;
 use Behat\Behat\Hook\Scope\BeforeScenarioScope;
 use TestHelpers\HttpRequestHelper;
+use PHPUnit\Framework\Assert;
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/ChecksumContext.php]: this is not needed now

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/ChecksumContext.php]: done :rocket:

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/22205/47/12
@kiranparajuli589 - needs rebase to current core master

**Review by phil-davis — APPROVED** [review]

**Pull request merged**